packages/flutter_test/lib/src/finders.dart

```diff
@@ -469,3 +483,24 @@ class _ElementPredicateFinder extends MatchFinder {
     return predicate(candidate);
   }
 }
+
+class _DescendantFinder extends Finder {
+  _DescendantFinder(this.ancestor, this.descendant, { bool skipOffstage: true }) : super(skipOffstage: skipOffstage);
+
+  final Finder ancestor;
+  final Finder descendant;
+
+  @override
+  String get description => '${ancestor.description} has descendant(s) with ${descendant.description} ';
+
+  @override
+  Iterable<Element> apply(Iterable<Element> candidates) {
+    return candidates.where((Element element) => descendant.evaluate().contains(element));
+  }
+
+  @override
+  Iterable<Element> get _allElements {
```

Comment:
Yeah, I think exposing this makes sense. Let's rename it to something reasonable though, like "candidates", since it's no longer all the elements. (if you pick "candidates" itself, maybe rename the arguments that already use that name to something else... not sure what).